test/unit/lodash.spec.js

```diff
@@ -0,0 +1,275 @@
+                if (index !== 3) {
+                    expect(_.includes.apply(_, [element, predicate])).to.eql(true);
+                } else {
+                    return false;
+                }
+            });
+            expect(_.includes.apply(_, [array[3], predicate])).to.eql(false);
+        });
+    });
+
+    describe('map', function () {
+        it('should map one of the property name', function () {
+            var testFunction = function () {
+                if (Math.random() < 0.2) {
+                    return {firstName: 'Ali', lastName: 'BOM', user: false};
+                } else if (Math.random() < 0.4) {
+                    return {firstName: 'Kowal', lastName: 'Kot', user: true};
+                } else if (Math.random() < 0.6) {
+                    return {firstName: 'Buro', lastName: 'Fisher', user: true};
+                } else if (Math.random() < 0.8) {
+                    return {firstName: 'Barney', lastName: 'Sick', user: false};
+                } else {
+                    return {firstName: 'Draco', lastName: 'Malio', user: true};
+                }
+            };
+            var testArray = _.fill(new Array(50), 1).map(testFunction);
+            var testArray2 = _.fill(new Array(50), 1).map(testFunction);
+            var params = datasets.map(testArray);
+            var params2 = datasets.map(testArray2);
+            var array = params[0];
+            var result = _.map.apply(_, params);
+            var result2 = _.map.apply(_, params2);
+            expect(array.length).to.eql(50);
+            expect(_.isString(params[1])).to.eql(true);
+            expect(_.map.apply(_, datasets.map(testArray))).to.eql(result);
+            expect(_.map.apply(_, datasets.map(testArray2))).to.eql(result2);
+        });
+        it('should square only even numbers', function () {
+            var testFunction = function () {
+                var randomNumber = Math.floor(Math.random() * 60) + 1;
+                if (Math.random() < 0.3) {
+                    return randomNumber;
+                } else if (Math.random() < 0.6) {
+                    return randomNumber;
+                } else {
+                    return randomNumber;
+                }
+            };
+            var testArray = _.fill(new Array(20), 1).map(testFunction);
+            var testArray2 = _.fill(new Array(10), 1).map(testFunction);
+            var params = datasets.map2(testArray);
+            var array = params[0];
+            var result = _.map.apply(_, params);
+            expect(array.length).to.eql(20);
+            expect(_.map.apply(_, datasets.map2(testArray))).to.eql(result);
+            params = datasets.map2(testArray2);
+            array = params[0];
+            expect(array.length).to.eql(10);
+            result = _.map.apply(_, params);
+            expect(_.map.apply(_, datasets.map2(testArray2))).to.eql(result);
+        });
+    });
+
+    describe('partition', function () {
```

Comment:
partition: First exercise allows us to use different data than expected. I find it annoying having to copy all objects from tests just to provide them as collection(you should hardcore it somewhere or maybe even pass as a parameter?). As for the rest scenarios I see no bigger meaning for them, although they work.